DOCSP-46394: CRUD remaining sections (#83)
mongodb/docs-mongoid: source/interact-data/crud.txt

``reset`` method, as shown in the following code:

.. literalinclude:: /includes/interact-data/crud.rb
   :language: ruby
   :start-after: start-dirty-tracking-reset
   :end-before: end-dirty-tracking-reset

Persistence
~~~~~~~~~~~

{+odm+} uses dirty tracking as the basis of all persistence operations.
It evaluates the changes on a document and atomically updates only what
has changed, compared to other frameworks that write the entire document on
each save. If you don't make any changes, {+odm+} does not access the
database when you call ``Model#save``.

View Previous Changes
~~~~~~~~~~~~~~~~~~~~~

After you persist a model to MongoDB, you can still see what changes
were made previously by calling the ``Model#previous_changes`` method,
as shown in the following code:

.. literalinclude:: /includes/interact-data/crud.rb
   :language: ruby
   :start-after: start-dirty-tracking-prev
   :end-before: end-dirty-tracking-prev

Update Container Fields
-----------------------

{+odm+} currently has an issue that prevents changes to attributes of
container types, such as ``Set`` or ``Array``, from saving to MongoDB.
You must assign all fields, including container types, for their values

> to save to MongoDB.

Author: Good idea! That could be a backlog ticket that we could file outside of the standardization. i can make it now

> to save to MongoDB.

Author: https://jira.mongodb.org/browse/DOCSP-46507

.. literalinclude:: /includes/interact-data/crud.rb
   :language: ruby
   :start-after: start-container-save
   :end-before: end-container-save

Read-only Documents
-------------------

You can mark documents as read-only in the following ways, depending on
the value of the ``Mongoid.legacy_readonly`` feature flag:

- If this flag is turned *off*, you can mark a document as
  read-only by calling the ``readonly!`` method on that document. The
  resulting read-only document raises a ``ReadonlyDocument`` error if
  you attempt to perform any persistence operation, including, but not
  limited to, saving, updating, deleting, and destroying. Note that
  reloading *does not* reset the read-only state.
  
  .. code-block:: ruby

     person = Person.first
     person.readonly? # => false
     person.readonly! # Sets the document as read-only
     person.readonly? # => true
     person.name = "Larissa Shay" # Changes the document
     person.save # => raises ReadonlyDocument error
     person.reload.readonly? # => true

- If this flag is turned ``on``, you can mark a document as read-only
  when that document is projected, such as by using ``only`` or

>   ``without``. The resulting read-only document is not deletable or
>   destroyable, as {+odm+} raises a ``ReadonlyDocument`` error, but it is
>   saveable and updatable. The read-only status *is reset* if you reload the

Reviewer: ```suggestion
  ``without``. As a result, you can't delete or destroy the read-only document,
  as {+odm+} raises a ``ReadonlyDocument`` error, but you can
  save and update it. The read-only status *is reset* if you reload the
```

  document.

  .. code-block:: ruby

     person = Person.only(:name).first
     person.readonly? # => true
     person.destroy # => raises ReadonlyDocument error
     person.reload.readonly? # => false

.. tip:: Projection

   To learn more about projections, see the
   :ref:`mongoid-data-projection` section of the Modify Query

>    Results guide.

Reviewer: Should this be indented to go under the second bullet?

> Override readonly?
> ~~~~~~~~~~~~~~~~~~
> 
> You can also make a document read-only by overriding the ``readonly?``
> method, as shown in the following code:

Reviewer: Since the previous section says: "You can make documents readonly in the following ways..." I think this can be a part of that section instead of having a dedicated header
```suggestion
You can also make a document read-only by overriding the ``readonly?``
method, as shown in the following code:
```
